### PR TITLE
[series/0.17] fix CI trigger (and build)

### PR DESCRIPTION
**Pull request by Baccata**
Fixes the blob pattern used by the CI trigger to fix workflows not running on PRs targeting the series/0.17 branch

closes https://github.com/disneystreaming/smithy4s/issues/504

#### Note about the large diff 

This is only due to the fact that the scalafmt.conf was off and some generated sources should not have been formatted, ever. 

#### Regarding change of assumed dependencies 

Avoid assuming dependencies during codegen (except for alloy)

* Only dependency assumed is alloy-core
* Assumption happens via build-tools default settings being set by
plugins/modules
* If users want to generate code from AWS, they need to manually
ascribe the dependencies containing AWS traits (updated the documentation in consequence)

Alloy can evolve independently from Smithy4s, and we cannot assume the files it contains will remain at stable locations across versions, so hardcoding a reference to the smithy files contained by alloy artifacts would be ill advised. Likewise, we cannot force the use "discoverModels" because it can backfire easily. The only solution is to have the plugins force the fetching of alloy (by default) at the build-tool level.

#### Mitigating the problem 

The user-declared dependencies are now propagating transitively. This means that, internally to a project, users can declare a smithy dependency once, and downstream projects will see it too. It's not perfect but alleviate the pain a little. 

#### Alternative considered 

Over the past few days I've done a fair bit of research to see whether we could bake something that would involve modules tracking the dependencies their used during their code generation, so that downstream could use that information to save the users having to worry about where some traits came from. My conclusion is that, though this would be ideal and provide the best user experience, it would require a lot of fiddling in non-trivial build-tool-specific layers (implying duplication of efforts between mill and SBT). Essentially, we'd need to bake some extra information in poms issued by projects that have smithy4s plugins enabled, and handcraft some resolution logic (respecting each build tool semantics) to do yet another pass of resolution after resolving what the user stated. If that would be reasonably easy on mill, it's much harder on SBT that effectively hides coursier behind layers and layers of indirections and doesn't give us a handle to the information we need in the APIs it exposes (like, we can't get a reference to the raw pom.xml when using SBT's resolution mechanism)". 

So, though this problem is very much solvable, the benefit/cost ratio is too low to pull the trigger on this at this time. Requiring more information from the users who want to generate AWS specs, or compliance-tests, is more acceptable (at this point in time, anyway). 

#### Edit

Some folks gave me some hints that may make the alternative described above possible. If that's the case, it'll come in a later PR

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -995,15 +990,27 @@ def genSmithyImpl(config: Configuration) = Def.task {
...
+              val discoverModelsOpt =
+                (if (discoverModels) List("--discover-models") else Nil)
+              val allowedNsOpt =
+                if (allowedNS.isDefined)
+                  List("--allowed-ns", allowedNS.get.mkString(","))
+                else Nil
+              val skipOpt = skip.flatMap(s => List("--skip", s))
+              val dependenciesOpt =
+                if (smithy4sDependencies.nonEmpty)
+                  List("--dependencies", smithy4sDependencies.mkString(","))
```
Comment (Baccata) [inline build.sbt:1005]: added this option

File: modules/core/src-2/generated/kinds/functorK.scala [hunk 2]
```diff
@@ -19,58 +19,35 @@
 package smithy4s
 package kinds
 
-trait FunctorK[Alg[_[_]]] {
+trait FunctorK[Alg[_[_]]]{
```
Comment (Baccata) [inline modules/core/src-2/generated/kinds/functorK.scala:22]: just formatting changes

File: modules/core/src-2/generated/kinds/kinds.scala [hunk 3]
```diff
@@ -19,20 +19,17 @@
 package smithy4s
 package kinds
 
-object Kind1 {
+object Kind1{
```
Comment (Baccata) [inline modules/core/src-2/generated/kinds/kinds.scala:22]: just formatting changes

File: modules/core/src/generated/schema/PartiallyAppliedStruct.scala [hunk 4]
```diff
@@ -19,1174 +19,39 @@
 package smithy4s
 package schema
 
-class PartiallyAppliedStruct[S] protected[schema] (placeholder: ShapeId) {
+class PartiallyAppliedStruct[S] protected[schema](placeholder: ShapeId) {
```
Comment (Baccata) [inline modules/core/src/generated/schema/PartiallyAppliedStruct.scala:22]: just formatting changes in this file

File: build.sbt (hunk 1 above)
Comment (daddykotex) [reply build.sbt:1005]: I love that you got rid of most genDiscoverModels

**Comment by Baccata** [comment]
@kubukoz , I'm tagging this PR as closing this : https://github.com/disneystreaming/smithy4s/issues/504 : the `smithy-aws-traits` now has to provided explicitly by the users, but now propagates correctly to downstream project. 

I'm aware that the explicit providing is not ideal, but trying to automate the propagation through "maven boundaries" is way too involved for what I want to go in the next release.

File: modules/codegen/src/smithy4s/codegen/ModelLoader.scala [hunk 5]
```diff
@@ -25,82 +25,67 @@ import software.amazon.smithy.model.Model
...
+    // Loading the upstream model
+    val upstreamModel = Model
       .assembler()
+      // disabling cache to support snapshot-driven experimentation
+      .putProperty(ModelAssembler.DISABLE_JAR_CACHE, true)
       .addClasspathModels(currentClassLoader, discoverModels)
+      .addImports(modelsInJars)
       .assemble()
       .unwrap()
-      .toBuilder()
```
Comment (Baccata) [inline modules/codegen/src/smithy4s/codegen/ModelLoader.scala:54]: this logic is made simpler by the fact that we're removing the assumption that (alloy) models are communicated through the classpath. The plugins (SBT, mill, CLI) are now responsible for setting the dependency to alloy as default.

File: modules/core/src-3/generated/kinds/kinds.scala [hunk 6]
```diff
@@ -19,21 +19,17 @@
 package smithy4s
 package kinds
 
-object Kind1 {
+object Kind1{
```
Comment (Baccata) [inline modules/core/src-3/generated/kinds/kinds.scala:22]: just formatting changes

File: modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sModule.scala [hunk 7]
```diff
@@ -62,8 +70,12 @@ trait Smithy4sModule extends ScalaModule {
   def smithy4sSmithyLibrary: T[Boolean] = true
 
   def smithy4sTransitiveIvyDeps: T[Agg[Dep]] = T {
-    smithy4sIvyDeps() ++ T
-      .traverse(moduleDeps)(_.transitiveIvyDeps)()
```
Comment (Baccata) [inline modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sModule.scala:66]: was not recursively calling the correct task there

File: modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sModule.scala [hunk 8]
```diff
@@ -62,8 +70,12 @@ trait Smithy4sModule extends ScalaModule {
...
+    smithy4sAllDeps() ++ T
+      .traverse(moduleDeps) {
+        case m if m.isInstanceOf[Smithy4sModule] =>
```
Comment (daddykotex) [inline modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sModule.scala:75]: interesting

**Comment by daddykotex** [comment]
> I'm satisfied with the content of the PR, but I'll build it locally and update some example projects that I have locally and see if anything breaks in unexpected ways

Works fine for a dummy project, made me realize that we have to update: https://github.com/disneystreaming/smithy4s.g8/pull/14

**Review by lewisjkl — APPROVED** [review]

**Pull request merged**